File: auto.py
class Auto:
    def __init__(self, marca, modelo, color, motor=None, cantidad_puertas=None, tipo_traccion=None, anno_fabricacion=None):
        self.marca = marca
        self.modelo = modelo
        self.color = color
        self.velocidad = 0
        self.motor = motor
        self.cantidad_puertas = cantidad_puertas
        self.tipo_traccion = tipo_traccion
        self.anno_fabricacion = anno_fabricacion

    def acelerar(self, incremento):
        self.velocidad += incremento
        print(f"El auto ahora va a {self.velocidad} km/h.")

    def frenar(self, decremento):
        self.velocidad -= decremento
        print(f"El auto ahora va a {self.velocidad} km/h.")

    def set_motor(self, motor):
        self.motor = motor

    def get_motor(self):
        return self.motor

    def set_color(self, color):
        self.color = color

    def get_color(self):
        return self.color

    def set_cantidad_puertas(self, cantidad_puertas):
        self.cantidad_puertas = cantidad_puertas

    def get_cantidad_puertas(self):
        return self.cantidad_puertas

    def set_tipo_traccion(self, tipo_traccion):
        self.tipo_traccion = tipo_traccion

    def get_tipo_traccion(self):
        return self.tipo_traccion

    def set_anno_fabricacion(self, anno_fabricacion):
        self.anno_fabricacion = anno_fabricacion

    def get_anno_fabricacion(self):
        return self.anno_fabricacion

File: auto.js
class Auto {
    constructor(marca, modelo, color, motor = null, cantidad_puertas = null, tipo_traccion = null, anno_fabricacion = null) {
        this.marca = marca;
        this.modelo = modelo;
        this.color = color;
        this.velocidad = 0;
        this.motor = motor;
        this.cantidad_puertas = cantidad_puertas;
        this.tipo_traccion = tipo_traccion;
        this.anno_fabricacion = anno_fabricacion;
    }

    acelerar(incremento) {
        this.velocidad += incremento;
        console.log(`El auto ahora va a ${this.velocidad} km/h.`);
    }

    frenar(decremento) {
        this.velocidad -= decremento;
        console.log(`El auto ahora va a ${this.velocidad} km/h.`);
    }

    setMotor(motor) {
        this.motor = motor;
    }

    getMotor() {
        return this.motor;
    }

    setColor(color) {
        this.color = color;
    }

    getColor() {
        return this.color;
    }

    setCantidadPuertas(cantidad_puertas) {
        this.cantidad_puertas = cantidad_puertas;
    }

    getCantidadPuertas() {
        return this.cantidad_puertas;
    }

    setTipoTraccion(tipo_traccion) {
        this.tipo_traccion = tipo_traccion;
    }

    getTipoTraccion() {
        return this.tipo_traccion;
    }

    setAnnoFabricacion(anno_fabricacion) {
        this.anno_fabricacion = anno_fabricacion;
    }

    getAnnoFabricacion() {
        return this.anno_fabricacion;
    }
}

// Exportar la clase para que pueda ser importada en otros archivos
module.exports = Auto;